### PR TITLE
feat: CLI display owner when listing CE workflow runs

### PR DESCRIPTION
**Pull request by BenjaminMummery**
# The problem

We get 'owner' back from the platform APU when listing workflows. At present this information is not displayed when listing workflow runs in the CLI.

Current output:

```
$ orq wf list...

Workflow Run ID   Status     Succeeded Tasks   Start Time                 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
wf1               SUCCEEDED  1/1               Fri Feb 24 08:26:07 2023   
wf2               WAITING    0/1                                          
```

# This PR's solution

Updates the presenter logic to add the `Owner` column if at least one of the workflow runs being listed has a populated owner field. Since we don't return owners for local runs, this has no effect when doing `orq wf list -c ray`, but for CE runs the output is now:

```
Workflow Run ID   Status     Succeeded Tasks   Start Time                 Owner
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
wf1               SUCCEEDED  1/1               Fri Feb 24 08:26:07 2023   evil/emiliano.zapata@zapatacomputing…
wf2               WAITING    0/1                                          evil/emiliano.zapata@zapatacomputing…
```

# Checklist

_Check that this PR satisfies the following items:_

- [X] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [X] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [X] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

[ORQSDK-924]

[ORQSDK-924]: https://zapatacomputing.atlassian.net/browse/ORQSDK-924?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/orquestra/sdk/_base/cli/_ui/_presenters.py [hunk 1]
```diff
@@ -362,19 +362,51 @@ def show_wf_run(self, summary: ui_models.WFRunSummary):
...
+            wf2               WAITING    0/1                                          evil/emiliano.zapata@zapatacomputing…
+
+        Args:
+            summary: A list of workflow run summaries to be displayed.
+        """  # noqa: E501
+        show_owner: bool = False
+        for row in summary.wf_rows:
+            if row.owner:
+                show_owner = True
+                break
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/cli/_ui/_presenters.py:391]: **suggestion (non-blocking, style):** If we want to use fancy one-liners xD
```suggestion
        show_owner = any(row.owner for row in summary.wf_rows)
```

File: src/orquestra/sdk/_base/cli/_ui/_presenters.py [hunk 2]
```diff
@@ -362,19 +362,51 @@ def show_wf_run(self, summary: ui_models.WFRunSummary):
...
+        if show_owner:
+            table.add_column(header="Owner")
+
         for run in summary.wf_rows:
             table.add_row(
                 run.workflow_run_id,
                 run.status,
                 run.tasks_succeeded,
                 _format_datetime(run.start_time),
+                run.owner,
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/cli/_ui/_presenters.py:409]: **question:** ‏If we don't show owners, what happens if we add extra column?

File: tests/cli/ui/data/list_wf_runs.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
- ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 
-  wf1               mocked status   x/y               Fri Feb 24 08:26:07 2023  
-  wf2               mocked status   x/y                                         
-                                                                                
+                                                                                   
+  Workflow Run ID   Status          Succeeded Tasks   Start Time                   
+ ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 
+  wf1               mocked status   x/y               Fri Feb 24 08:26:07 2023     
+  wf2               mocked status   x/y                                            
+
```
Comment (jamesclark-Zapata) [inline tests/cli/ui/data/list_wf_runs.txt:6]: **question:** ‏Is this whitespace change related to my previous question?

File: src/orquestra/sdk/_base/cli/_ui/_presenters.py [hunk 4]
```diff
@@ -362,19 +362,51 @@ def show_wf_run(self, summary: ui_models.WFRunSummary):
         self._console.print(self.get_wf_run(summary))
 
     def show_wf_list(self, summary: ui_models.WFList):
+        """Display a list of workflow runs.
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/cli/_ui/_presenters.py:365]: **praise:** ‏Nice docstring!

**Review by jamesclark-Zapata — APPROVED** [review]
lgtm.

1 non-blocking suggestion and two related questions, in addition to what I asked on Slack. Overall, these don't block the merge imo.

File: tests/cli/ui/data/list_wf_runs.txt (hunk 3 above)
Comment (BenjaminMummery) [reply tests/cli/ui/data/list_wf_runs.txt]: Aah yeah, good catch that's exactly what's happening! Empty column removed in [e552428](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/320/commits/e552428122a02399f6d87cae710bd93b436689db)

**Comment by jamesclark-Zapata** [comment]
lgtm!
